src/getPage.tsx

```diff
@@ -93,15 +119,21 @@ export default async function getPage({
 
   // Optionally wrap with custom Document
   if (useDocument) {
-    pageElement = await renderDocument({
-      pageElement,
+    serverPageElement = await renderDocument({
+      pageElement: serverPageElement,
       options,
       pageObject,
       pageData,
     });
   }
 
+  const html: string = ReactDOMServer.renderToStaticMarkup(
```

Comment:
We already `renderToStaticMarkup` in here: https://github.com/toomuchdesign/next-page-tester/blob/master/src/_document/render.tsx#L45. Cant we reuse that? 

Moreover, I think if we make this work flawlessly, I don't see a reason why `useDocument` and `useApp` would stay configuration options. They just complicate the code and increase the API surface while perhaps keeping users confused (as they would expect the whole page to render)?

Reply:
1- If we always rendered a `_document` we could always end up rendering the tree as string with `renderDocument`. Do we ever call that `renderPage` method or it's something we provide to Next.js internals? Maybe we could be able to end up with an implementation closer to Next.js, since now we should only render `_document` as a string, right?

2- I'd say yes, even though ignoring `_app` and _document` rendering in complex applications could help leaving unnecessary mocks/setups out of tests (?).

If it worked it would also allow to get rid of most of our `testHelpers.ts` setup. Let's see how it evolves.